samcli/local/lambda_service/local_lambda_invoke_service.py

```diff
@@ -46,14 +48,65 @@ def create(self):
                                methods=['POST'],
                                provide_automatic_options=False)
 
+        # setup request validation before Flask calls the view_func
+        self._app.before_request = self.validate_request
+
         self._construct_error_handling()
 
+    def validate_request(self):
+        """
+        Validates the incoming request
+
+        Returns
+        -------
+        flask.Response
+            A Response is returned in the following cases:
+            1. The Request data is not json serializable
+            2. Query Parameters are sent to the endpoint
+            3. The Request Content-Type is not application/json
+
+            Otherwise this function does not return anything
+        """
+        flask_request = request
+        request_data = flask_request.get_data()
+
+        if not request_data:
+            request_data = b'{}'
+
+        request_data = request_data.decode('utf-8')
+
+        try:
+            json.loads(request_data)
+        except ValueError as json_error:
+            LOG.debug("Request body was not json.")
+            return LambdaErrorResponses.invalid_request_content(
+                "Could not parse request body into json: {}".format(str(json_error)))
+
+        if flask_request.args:
```

Comment:
This is a POST request. so query params are less of a problem. But anyway, good to handle.